app-modules/authorization/tests/Feature/Filament/Resources/User/RestoreUserTest.php

```diff
@@ -0,0 +1,157 @@
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use App\Models\User;
+
+use function Tests\asSuperAdmin;
+use function Pest\Laravel\actingAs;
+use function Pest\Livewire\livewire;
+
+use Filament\Tables\Actions\RestoreAction;
+use Filament\Tables\Filters\TrashedFilter;
+use App\Filament\Resources\UserResource\Pages\EditUser;
+use App\Filament\Resources\UserResource\Pages\ListUsers;
+use App\Filament\Resources\UserResource\Pages\CreateUser;
+
+it('show trashed filter only if user has user restore permission', function () {
+    $user = User::factory()->create();
+
+    $user->givePermissionTo('user.view-any', 'user.*.view', 'user.*.restore');
+    actingAs($user);
+
+    livewire(ListUsers::class)
+        ->assertTableFilterExists(TrashedFilter::class);
+});
+
+it('do not show soft deleted users when filter is not selected', function () {
+    asSuperAdmin();
+    $users = User::factory()->count(3)->create();
+
+    // Soft-delete one of the users
+    $softDeletedUser = $users->first();
+    $softDeletedUser->delete();
+    $nonDeletedUserRecords = User::get();
+    $softDeletedUserRecords = User::onlyTrashed()->get();
+
+    livewire(ListUsers::class)
+        ->removeTableFilters()
+        ->assertCanSeeTableRecords($nonDeletedUserRecords)
+        ->assertCanNotSeeTableRecords($softDeletedUserRecords);
+});
+
+it('can see soft deleted, non soft deleted records and status column only if the filter is on', function () {
+    asSuperAdmin();
+    $users = User::factory()->count(3)->create();
+
+    // Soft-delete one of the users
+    $softDeletedUser = $users->first();
+    $softDeletedUser->delete();
+
+    livewire(ListUsers::class)
+        ->filterTable(TrashedFilter::class)
+        ->assertTableColumnExists('deleted_at')
```

Comment:
You may want to assert `visible` rather than exists here. And you probably should test that the filter is not visible before you apply `filterTable`.

You may be able to do so fluently, but if not you can assign `livewire(ListUsers::class)` to a variable.

Reply:
@Orrison As discussed assertTableColumnVisible() not exists.